libpromises/syntax.c

```diff
@@ -889,9 +885,7 @@ bool CheckParseVariableName(const char *const name)
         if (count == 1)
         {
             // Check that there is something before and after first dot:
-            sscanf(name, "%[^.].%s", scopeid, vlval);
-
-            if (scopeid[0] == '\0' || vlval[0] == '\0')
+            if (name == first_dot || first_dot[1] == '\0')
```

Comment:
`name[0] == '.'` would be more understandable for me here.

Reply:
Alright.